### ISSUE TITLE
How can i create x12 messages from clr objects?

### ISSUE THREAD

**Issue by aronmek** [phatboyg/Machete#53]
First of all thanks for creating this project! It looks awesome!

I played around with the code, and it seems that the API is all tailored around parsing existing formatted messages into clr objects.
I'm wondering how to go about creating a well formatted  x12/hl7 message from clr types.

Is that possible?

**Comment by phatboyg** [comment]
That hasn't been a focus at this point, to be honest. Right now, it's around parsing and processing data - the primary use case is reading data from systems for analytics purposes, versus generating valid HL7 or X12 output. The formatting will come around eventually, as will the ability to create the components, segments, etc. but that isn't where most of the work is being done at present.

@ahives can give more details around that aspect.

**Comment by ahives** [comment]
So, the eventual goal is to be able to do this. But as @phatboyg said our current focus really is on getting the parsing engine solid, then we will build out from that. We have a few open issues around Layouts, which are basically POCOs. Once this work is done we will beef up the formatting and translator functionality. Some of that is already done, but, it needs to be extended to support POCOs (i.e. Layouts, segments, etc.). Once done though, you will have the ability to create your own POCOs (Layouts, segments, components, etc.) using the Translate API and then format it to text, which would give you a string HL7/X12 message. This will definitely be there when we rev the package to stable as it is a core use case.

**Comment by aronmek** [comment]
Thanks @phatboyg and @ahives for your prompt reply.
I guess I'll be waiting around until I can start to use it for this purpose.

Thanks again!
